internal/backends/mysql/collection.go

```diff
@@ -40,27 +51,317 @@ func newCollection(r *metadata.Registry, dbName, name string) backends.Collectio
+			if err != nil {
+				return lazyerrors.Error(err)
+			}
+
+			if _, err = tx.ExecContext(ctx, q, args...); err != nil {
+				var mysqlErr *mysql.MySQLError
+				if errors.As(err, &mysqlErr) && mysqlErr.Number == ErrDuplicateEntry {
+					return backends.NewError(backends.ErrorCodeInsertDuplicateID, err)
+				}
+
+				return lazyerrors.Error(err)
+			}
+		}
+
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return new(backends.InsertAllResult), nil
 }
 
 // UpdateAll implements backend.Collection interface.
 func (c *collection) UpdateAll(ctx context.Context, params *backends.UpdateAllParams) (*backends.UpdateAllResult, error) {
-	return nil, lazyerrors.New("not yet implemented")
+	p, err := c.r.DatabaseGetExisting(ctx, c.dbName)
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	var res backends.UpdateAllResult
+	if p == nil {
+		return &res, nil
+	}
+
+	meta, err := c.r.CollectionGet(ctx, c.dbName, c.name)
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	if meta == nil {
+		return &res, nil
+	}
+
+	q := fmt.Sprintf(
+		`UPDATE %s.%s SET %s = ? WHERE %s = ?`,
+		c.dbName, meta.TableName,
+		metadata.DefaultColumn,
+		metadata.IDColumn,
+	)
+
+	err = p.InTransaction(ctx, func(tx *fsql.Tx) error {
+		for _, doc := range params.Docs {
+			var b []byte
+
+			b, err = sjson.Marshal(doc)
+			if err != nil {
+				return lazyerrors.Error(err)
+			}
+
+			id, _ := doc.Get("_id")
+			must.NotBeZero(id)
+
+			arg := must.NotFail(sjson.MarshalSingleValue(id))
+
+			var stats sql.Result
+
+			stats, err = tx.ExecContext(ctx, q, b, arg)
+			if err != nil {
+				return lazyerrors.Error(err)
+			}
+
+			var ra int64
+
+			ra, err = stats.RowsAffected()
+			if err != nil {
+				return lazyerrors.Error(err)
+			}
+
+			res.Updated += int32(ra)
+		}
+
+		return nil
+	})
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	return &res, nil
 }
 
 // DeleteAll implements backend.Collection interface.
 func (c *collection) DeleteAll(ctx context.Context, params *backends.DeleteAllParams) (*backends.DeleteAllResult, error) {
-	return nil, lazyerrors.New("not yet implemented")
+	p, err := c.r.DatabaseGetExisting(ctx, c.dbName)
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	if p == nil {
+		return &backends.DeleteAllResult{Deleted: 0}, nil
+	}
+
+	meta, err := c.r.CollectionGet(ctx, c.dbName, c.name)
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	if meta == nil {
+		return &backends.DeleteAllResult{Deleted: 0}, nil
+	}
+
+	var column string
+	var placeholders []string
+	var args []any
+
+	if params.RecordIDs == nil {
+		placeholders = make([]string, len(params.IDs))
+		args = make([]any, len(params.RecordIDs))
+
+		for i, id := range params.RecordIDs {
+			placeholders[i] = "?"
+			args[i] = id
+		}
+
+		column = metadata.RecordIDColumn
+	}
+
+	q := fmt.Sprintf(
+		`DELETE FROM %s.%s WHERE %s IN (%s)`,
```

Comment:
Can `c.dbName` and `meta.Table` be escaped or sanitized? It may contain malicious query from user. PostgreSQL sanitize it with `pgx.Identifier{c.dbName, meta.TableName}.Sanitize()`, and SQLite use `%q` to quote the table name. This comment applies to all other places too :hugs:

Reply:
I'll fix this in another PR instead.

Reply:
Yes sure :+1: